### PR TITLE
Migrations [Closes #22]

### LINKED ISSUE #22: Add database migrations
As a developer, I want to be able to specify database migrations for use during deployment, so that I can safely create and revert database updates.
- Should integrate with Sequelize
- Should allow data migrations as well

### PR DESCRIPTION
**Pull request by Jusdev89**
- Installed sequelize-cli
- Installed db
- Ran sequelize init command which created config/config.json file
- Created a migrations folder with three files for items, users and verification codes
- Successfully ran and tested db:migrate, db:migrate:undo, and db:migrate:all commands
- [Documentation](https://www.npmjs.com/package/sequelize-cli)

### CODE REVIEW

**Comment by shakalee14** [comment]
@jrob8577 We had trouble squashing these :( I @Moniarchy can't tell if these other commits have actually been squashed but I don't think they have been. If this is a problem, we can keep trying!

File: package.json [hunk 1]
```diff
@@ -10,13 +10,19 @@
     "start": "nf start",
     "lint": "eslint --color *.js; exit 0;",
     "test:watch": "mocha -w",
-    "test": "istanbul cover -x *_[sS]pec.js _mocha  -- -R spec"
+    "test": "istanbul cover -x *_[sS]pec.js _mocha  -- -R spec",
+    "migrate": "node_modules/.bin/sequelize db:migrate",
```
Comment (jrob8577) [inline package.json]: You do not need to include `node_modules/.bin/` - npm takes care of figuring out what thing we want to execute...

File: package.json [hunk 2]
```diff
@@ -10,13 +10,19 @@
...
+    "migrate:undo": "node_modules/.bin/sequelize db:migrate:undo",
+    "migrate:undo:all": "node_modules/.bin/sequelize db:migrate:undo:all",
```
Comment (jrob8577) [inline package.json]: I don't understand the distinction between the `undo` and `undo:all` - can you please include information in an appropriate location that documents these actions?

File: package.json [hunk 3]
```diff
@@ -10,13 +10,19 @@
...
+    "migrate:test": "node_modules/.bin/sequelize db:migrate --url "
```
Comment (jrob8577) [inline package.json]: What does this test?  Can you please include information in an appropriate location that documents thus command?

File: package.json [hunk 4]
```diff
@@ -10,13 +10,19 @@
...
   },
   "dependencies": {
     "bcrypt": "^0.8.7",
     "body-parser": "~1.15.1",
     "cookie-parser": "~1.4.3",
+    "db": "^1.0.6",
```
Comment (jrob8577) [inline package.json]: Is this a dependency required for the migrations?  Is it explicitly required?

File: migrations/verify-migrations.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+const { VerificationCode }  = require('../database/models')
+
+module.exports = {
+  up: ( migration, DataTypes ) => migration.createTable( VerificationCode.tableName, VerificationCode.attributes ) ,
```
Comment (jrob8577) [inline migrations/verify-migrations.js]: Please keep lines less than 80 characters - I'll add this to CONTRIBUTING.md

File: migrations/item-migrations.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+const { Item }  = require('../database/models')
+
+module.exports = {
+  up: ( migration, DataTypes ) => migration.createTable( Item.tableName, Item.attributes ),
```
Comment (jrob8577) [inline migrations/item-migrations.js]: Please keep lines less than 80 characters - I'll add this to CONTRIBUTING.md

File: migrations/user-migrations.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+'use strict'
+
+const { User }  = require('../database/models')
+
+module.exports = {
+  up: ( migration, DataTypes ) => migration.createTable( User.tableName, User.attributes ),
```
Comment (jrob8577) [inline migrations/user-migrations.js]: Please keep lines less than 80 characters - I'll add this to CONTRIBUTING.md

File: README.md [hunk 8]
```diff
@@ -39,3 +39,13 @@ Floworky uses flow for static type checking, which can be manually run using `np
 
 ### Resources
 See [RESOURCES.md](RESOURCES.md)
+
+### Migrations
```
Comment (jrob8577) [inline README.md]: This doesn't feel like the right section for this...  We may need to introduce a new file that lists developer specific information, or add this into an appropriate section in CONTRIBUTING.md

**Comment by Jusdev89** [comment]
@jrob8577 we addressed all of your request.

File: README.md [hunk 9]
```diff
@@ -39,3 +39,6 @@ Floworky uses flow for static type checking, which can be manually run using `np
 
 ### Resources
 See [RESOURCES.md](RESOURCES.md)
+
```
Comment (jrob8577) [inline README.md]: Removing unneeded whitespace.

File: migrations/item-migrations.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+'use strict'
+
+const { Item }  = require('../database/models')
+
+module.exports = {
+  up: ( migration, DataTypes ) => {
+    return migration.createTable( Item.tableName, Item.attributes )
+  },
```
Comment (jrob8577) [inline migrations/item-migrations.js]: omit braces and `return` statement for single line functions (this applies elsewhere in this commit)

File: package.json [hunk 11]
```diff
@@ -10,13 +10,18 @@
     "start": "nf start",
     "lint": "eslint --color *.js; exit 0;",
     "test:watch": "mocha -w",
-    "test": "istanbul cover -x *_[sS]pec.js _mocha  -- -R spec"
+    "test": "istanbul cover -x *_[sS]pec.js _mocha  -- -R spec",
+    "migrate": "sequelize db:migrate",
+    "migrate:undo:all": "sequelize db:migrate:undo:all",
```
Comment (jrob8577) [inline package.json]: I don't think you need to remove the `migrate:undo` - I just don't know what it does.  This needs to be documented somewhere.  I still don't understand what the `migrate:test` command does either.

File: package.json [hunk 12]
```diff
@@ -10,13 +10,18 @@
...
+    "migrate:test": "sequelize db:migrate --url "
```
Comment (jrob8577) [inline package.json]: I did not know until reading the notes again that this requires an additional parameter - is there a way to obtain this from `process.env`?

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -15,3 +15,11 @@
 * No linting errors
 * Spaces (2), not tabs
 * Flow Type Annotations are included
+
+### Migrations
+* [Sequelize-Cli List of Commands](https://www.npmjs.com/package/sequelize-cli)
```
Comment (jrob8577) [inline CONTRIBUTING.md]: I think this needs more detail - I like the list of resources, but as a new developer on this project, I don't know what I need to do in order to create a migration... when is one necessary?  

As well, let's add a few bullets to the section above:
```
* When schema updated are required, sequelize schema migrations are provided with instructions to run in the PR message
* When data migrations are required, sequelize data migrations are provided with instructions to run in the PR message
```

File: package.json (hunk 11 above)
Comment (jrob8577) [inline package.json]: As well, I need to understand where in the deployment process these scripts ought to be run?

File: package.json [hunk 14]
```diff
@@ -10,13 +10,19 @@
     "start": "nf start",
     "lint": "eslint --color *.js; exit 0;",
     "test:watch": "mocha -w",
-    "test": "istanbul cover -x *_[sS]pec.js _mocha  -- -R spec"
+    "test": "istanbul cover -x *_[sS]pec.js _mocha  -- -R spec",
+    "migrate": "sequelize db:migrate",
+    "migrate:undo": "sequelize db:migrate:undo",
+    "migrate:undo:all": "sequelize db:migrate:undo:all",
+    "migrate:test": "sequelize db:migrate --url postgres://whoami@localhost:5432/floworky-test"
```
Comment (jrob8577) [inline package.json]: please remove this script - not sure what it does?

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -5,7 +5,10 @@
 * Create a topic branch.  Branch names should begin with the issue number, followed by text describing the work.  Please use lower-kebab-case (i.e. `2-readme-updated`).
 * Complete work
 * Submit a pull request
-
+  * If data must be migrated, create a data migration. See below for example:
```
Comment (jrob8577) [inline CONTRIBUTING.md]: Let's outdent this and change the text:
```
* Necessary migrations have been created with the following commands:
```

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -5,7 +5,10 @@
...
+  * Model: ```sequelize model:create --name TableName --attributes email:string,password:string```
+  * Migration: ```sequelize migrations:create --name add-email-verified-to-users```
+  * Updates or Creates database schema: ```npm run migrate```
```
Comment (jrob8577) [inline CONTRIBUTING.md]: Remove this line

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -5,7 +5,10 @@
 * Create a topic branch.  Branch names should begin with the issue number, followed by text describing the work.  Please use lower-kebab-case (i.e. `2-readme-updated`).
 * Complete work
 * Submit a pull request
-
+  * If data must be migrated, create a data migration. See below for example:
+  * Model: ```sequelize model:create --name TableName --attributes email:string,password:string```
```
Comment (jrob8577) [inline CONTRIBUTING.md]: Change this to:
```
`[sequelize model:create](insert documentation link here)`
```

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -5,7 +5,10 @@
...
+  * Migration: ```sequelize migrations:create --name add-email-verified-to-users```
```
Comment (jrob8577) [inline CONTRIBUTING.md]: Change this to:
```
`[sequelize migrations:create](insert documentation link here)`
```

**Pull request merged**